journal/fs.go

```diff
@@ -108,14 +112,30 @@ func (f *fsJournal) rollJournalFile() error {
 	if f.fi != nil {
 		_ = f.fi.Close()
 	}
+	end := build.Clock.Now()
+	current := filepath.Join(f.dir, "lotus-journal.ndjson")
+	rolled := filepath.Join(f.dir, fmt.Sprintf(
+		"lotus-journal-%s-%s.ndjson",
+		f.start.Format(RFC3339nocolon), end.Format(RFC3339nocolon),
```

Comment:
This breaks when we restart, because `f.start` will no longer have the original start timestamp, so the stamping would be inaccurate. We could stamp the file with the start timestamp, and then append the end timestamp on rotation, but that would make the `current` filename dynamic again. I'm OK just accepting the semantic change to stamp the rotated file with the end date.